segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/operations/StreamSegmentAppendOperation.java

```diff
@@ -34,11 +37,20 @@
 public class StreamSegmentAppendOperation extends StorageOperation implements AttributeUpdaterOperation, AutoCloseable {
     //region Members
 
+    // Represents that no hash has been computed for a given Append contents.
+    public static final long NO_HASH = Long.MIN_VALUE;
+
     private static final long NO_OFFSET = -1;
     protected long streamSegmentOffset;
     protected BufferView data;
     protected AttributeUpdateCollection attributeUpdates;
 
+    // Hash of the Append contents. Note that this is only used internally within the Segment Store to
+    // check data integrity and should not be considered for serialization (i.e., transient field).
+    @Getter
+    @Setter
+    private transient volatile long contentHash = NO_HASH;
```

Comment:
This should be final not volatile. If this is volatile it may not be consistent with the value in the data field. Also 'volatile' may not work properly with longs.

Reply:
Right, removed the `volatile` keyword as it may be problematic with `long` and may not be needed (is set only once and never modified).